# Organization_and_annotation_of_eukaryote_genomes
In this repository, you will find all the code to reproduce my analysis for the genome annotation of an eukaryotic genome lecture. Datasets used are from PacBio HiFi reads (WGS) and Illumina reads (RNA-seq) from the previous lecture, available on the [annotation and assembly repository](../../../../cr0sscheck/Annotation-and-assembly-course).

## Steps  
# First step: Annotation and classifiction of the transposable elements

. Annotation of transposable elements (TE) with EDTA
. Visualization and comparison of TE annotations generated by EDTA (using R)
. Refinement of TE classification using TEsorter
. Generation and visualization of TE dynamics (with R)

# Second step: Genes annotation with the MAKER pipeline

. Ab initio gene prediction with Augustus and GeneMark
. Integration of RNA-seq data to improve the accuracy of ab initio gene models
. Protein homology-based annotation with InterProScan
. Refinement of final gene models to ensure biologically consistent annotations
. Evaluation of gene annotation quality using BUSCO
. Visualization of gene annotations with Geneious

# Third step is the orthology based gene functional annotation and genome comparisons

. Search for homology with functionally validated proteins using the UniProt and TAIR10 databases
. Comparative genomics analysis with OrthoFinder
. Synteny analysis and visualization using GENESPACE (under R)
. Visualization of results, including summary statistics, dot plots, and synteny maps (also known as Riparian plots)
